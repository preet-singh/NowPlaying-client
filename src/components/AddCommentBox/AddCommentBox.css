@media screen and (min-width: 320px) {
  .add-comment-box {
    display: flex;
    flex-direction: column;
    padding-bottom: 40px;
  }
  
  .add-comment-box textarea {
    display: block;
    margin: auto;
    height: 5em;
    min-width: 80%;
    max-width: 80%;
  }

  .reaction-buttons button {
    color: white;
    background-color: transparent;
    border: 1px solid white;
    margin-left: 10px;
    margin-right: 10px;
    background-color: transparent;
    margin-top: 10px;
    width: 30px;
    height: 30px;
  }

  .add-comment-form > button {
    background-color: transparent;
    border: none !important;
    color: white;
    padding: 2px 10px 2px 10px;
    margin-top: 10px;
    float: right;
    margin-right: 21px;
  }

}

@media screen and (min-width: 768px){
  
  .add-comment-box textarea {
    display: block;
    margin: auto;
    height: 5em;
    min-width: 80%;
    max-width: 80%;
    min-height: 200px;
  }

  .reaction-buttons {
    display: flex;
    flex-direction: column;
    position: absolute;
    top: 85px;
    left: 5px;
  }

  .reaction-buttons button:hover {
    cursor: pointer;
  }

}