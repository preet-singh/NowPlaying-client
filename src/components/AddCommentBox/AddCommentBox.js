import React from 'react';
import './AddCommentBox.css';
import comment from '../Images/comment.svg'
import AuthApiService from '../../utils/auth-service';
import UserContext from '../../utils/context';

class AddCommentBox extends React.Component {
  static contextType = UserContext;

  state = {
    comment: '',
    error: null,
    happenings: []
  }

  componentDidMount() {
    AuthApiService.getHappeningEvents()
      .then(data => {
        this.setState({
          happenings: data
        })
      })
      .catch(res => {
        this.setState({
          error: res.error
        })
      })
  }

  handleCommentInput = text => {
    this.setState({
      comment: text,
    })
  }

  refreshCommentsCheck = async () => {
    let mediaType = this.props.category;
    if(mediaType === 'books'){
      let res = await AuthApiService.getBookComments(mediaType.thread, this.context.playingID).then(res => res);
      const orderedComments = res.sort((a,b) => a.comment_timestamp - b.comment_timestamp)
      this.context.setCurrentThreadComments(orderedComments);
    } 
    else if(mediaType === 'movies'){
      let res = await AuthApiService.getMovieComments(mediaType.thread, this.context.playingID).then(res => res);
      console.log(res);
      const orderedComments = res.sort((a,b) => a.comment_timestamp - b.comment_timestamp)
      await this.context.setCurrentThreadComments(orderedComments);
    }
    else if(mediaType === 'podcasts'){
      let res = await AuthApiService.getPodcastComments(mediaType.thread, this.context.playingID).then(res => res);
      const orderedComments = res.sort((a,b) => a.comment_timestamp - b.comment_timestamp)
      await this.context.setCurrentThreadComments(orderedComments);
    }
    else if(mediaType === 'tv_shows'){
      let res = await AuthApiService.getTVShowComments(mediaType.thread, this.context.playingID).then(res => res);
      const orderedComments = res.sort((a,b) => a.comment_timestamp - b.comment_timestamp)
      this.context.setCurrentThreadComments(orderedComments);
    }
  }

  handleCommentSubmit = async e => {
    e.preventDefault()
    let category = this.props.category.slice(0, this.props.category.length - 1)

    const commentBody = {
      user_comment: this.state.comment,
      comment_timestamp: this.context.mediaTimer,
      media_id: this.props.mediaId,
    }

    await AuthApiService.postComment(category, commentBody)
    await this.refreshCommentsCheck();

    await AuthApiService.getSpecificEvent(this.props.category, this.props.mediaId)
      .then(res => {
        console.log(res)
        const commentBodyHappenings = {
          username: this.context.user.username,
          user_comment: this.state.comment,
          media_title_comments: res[0].title,
          media_type: this.context.category,
          media_id: this.props.mediaId,
        }
        AuthApiService.postCommentHappenings(commentBodyHappenings)
          .then(res => console.log(res))
      })
    this.setState({comment: '', })
  }

  handleReactions = async reaction => {
    console.log(reaction)
    let category = this.props.category.slice(0, this.props.category.length - 1)
    const commentBody = {
      user_comment: reaction,
      comment_timestamp: this.context.mediaTimer,
      media_id: this.props.mediaId,
    }
    await AuthApiService.postComment(category, commentBody)
    await this.refreshCommentsCheck();
  }

  render() {
    return (
      <div className='add-comment-box'>
        <form className='add-comment-form' onSubmit={e => this.handleCommentSubmit(e)}>
          <textarea id='comment-text-input' type='text' value={this.state.comment} onChange={e => this.handleCommentInput(e.target.value)} />
          <button id='send-comment' type='submit'>Send</button>
        </form>
        {this.context.displayCommentBox ? <button id='close_commentBox' onClick={() => this.context.displayCommentSection()}>X</button> : null}
        <div className={this.context.playing ? 'reaction-buttons' : 'reaction-buttons_notPlaying'}>
          <button className='reaction-button' type='button' value='ðŸ˜‚' onClick={e => this.handleReactions(e.target.value)}><span role='img' aria-label='laughing emoji'>ðŸ˜‚</span></button>
          <button className='reaction-button' type='button' value='ðŸ˜”' onClick={e => this.handleReactions(e.target.value)}><span role='img' aria-label='laughing emoji'>ðŸ˜”</span></button>
          <button className='reaction-button' type='button' value='ðŸ˜±' onClick={e => this.handleReactions(e.target.value)}><span role='img' aria-label='laughing emoji'>ðŸ˜±</span></button>
        </div>
      </div>
    )
  }
}

export default AddCommentBox;