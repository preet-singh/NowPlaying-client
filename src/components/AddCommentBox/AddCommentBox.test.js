import React from 'react';
import ReactDOM from 'react-dom';
import AddCommentBox from './AddCommentBox';
import { MemoryRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

describe('Add comment box testing', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <MemoryRouter>
            <AddCommentBox  />
        </MemoryRouter>, 
    div);
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('renders the user comment box', () => {
    const wrapper = shallow(<AddCommentBox  />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
  
});