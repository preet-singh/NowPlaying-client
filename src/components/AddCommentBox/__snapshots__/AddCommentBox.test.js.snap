// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add comment box testing renders the user comment box 1`] = `
<div
  className="add-comment-box"
>
  <form
    className="add-comment-form"
    onSubmit={[Function]}
  >
    <textarea
      disabled={true}
      id="comment-text-input"
      onChange={[Function]}
      placeholder="Log in to enter a comment"
      type="text"
      value=""
    />
    <div
      className="reaction-buttons_notPlaying"
    >
      <button
        className="reaction-button"
        onClick={[Function]}
        type="button"
        value="1"
      >
        <span
          aria-label="laughing emoji"
          role="img"
        >
          ðŸ˜‚
        </span>
      </button>
      <button
        className="reaction-button"
        onClick={[Function]}
        type="button"
        value="2"
      >
        <span
          aria-label="sad emoji"
          role="img"
        >
          ðŸ˜”
        </span>
      </button>
      <button
        className="reaction-button"
        onClick={[Function]}
        type="button"
        value="3"
      >
        <span
          aria-label="shocked emoji"
          role="img"
        >
          ðŸ˜±
        </span>
      </button>
    </div>
    <button
      className="send-comment"
      type="submit"
    >
      Send
    </button>
  </form>
</div>
`;
